### ISSUE TITLE
Uncaught TypeError: Cannot read property 'target' of undefined

### ISSUE THREAD

**Issue by rickyp-uber** [uber/react-digraph#75]
Version information

Attempted to upgrade from React 16.4.2 to 16.5.2 and I seem to be getting a new error:
```
Uncaught TypeError: Cannot read property 'target' of undefined
 - GraphView.arrowClicked
and
Uncaught TypeError: Cannot read property 'which' of undefined
 - GraphView.handleNodeMouseLeave
```

Here is the useful parts of the package.lock file.
```
    "react": "^16.5.2",
    "react-digraph": "^4.1.0",
    "react-dom": "^16.5.2",
```

**Comment by rickyp-uber** [comment]
Looks like the code seems to be referencing an event of which it doesn't have access to.
```
  handleNodeMouseLeave = (d) => {
    const fromMouseup = event.which == 1;
```

This will cause issues with uglify and other libraries.

**Comment by TeslaFreak** [comment]
This same issue happens in handleSvgClicked as well

> TypeError: Cannot read property 'target' of undefined
>     at HTMLDivElement.GraphView._this.handleSvgClicked

```
if (!this.props.readOnly && d3.event.shiftKey) {
          var xycoords = d3.mouse(event.target);
          this.props.onCreateNode(xycoords[0], xycoords[1]);
          this.renderView();
        }
```

were all these events meant to be d3.event?

**Comment by iamsoorena** [comment]
this issue is the same as [this one.](https://github.com/uber/react-digraph/issues/80)

**Comment by rickyp-uber** [comment]
What version of D3 is everyone using here? I have 4.1.1 directly referenced in my package.json and I'm curious if that's what is causing the issue. Anyone using 5 (which is what the package.json of this library uses)?

**Comment by iamsoorena** [comment]
@rickyp-uber It's not about the D3 version.

**Comment by TeslaFreak** [comment]
Theres literally a pull request (#39) that would solve this but the guy never signed the contributor agreement. If someone just went in and redid what he did this would be resolved

**Comment by TeslaFreak** [comment]
@iamsoorena much appreciated, you're the hero this city needs

**Comment by ajbogh** [comment]
@rickyp-uber I've merged #82, if this is fixed in v4.1.7, then please feel free to resolve this issue. We may have to make another PR to the master branch if this happens there as well.

**Comment by mohammad141** [comment]
You can fix it by doing like this

function handleChnage(event) {
    const value = event.target.value;
}

<input onChange={handleChange} />

You should not do like this

<input onChange={handleChange()} />
